prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2596,13 +2643,16 @@ int rxr_endpoint(struct fid_domain *domain, struct fi_info *info,
 	rxr_ep->efa_outstanding_tx_ops = 0;
 	rxr_ep->shm_outstanding_tx_ops = 0;
 
-	ret = efa_cq_open(&efa_domain->util_domain.domain_fid, &cq_attr,
-			  &rxr_ep->rdm_cq, rxr_ep);
-	if (ret)
-		goto err_close_shm_ep;
+	assert(!(rxr_ep->ibv_cq_ex));
+
+	rxr_ep->ibv_cq_ex = rxr_ep_create_ibv_cq_ex(rxr_ep, &cq_attr);
 
-	ret = fi_ep_bind(rxr_ep->rdm_ep, &rxr_ep->rdm_cq->fid,
-			 FI_TRANSMIT | FI_RECV);
+	if (!rxr_ep->ibv_cq_ex) {
+		EFA_WARN(FI_LOG_CQ, "Unable to create extended CQ: %s\n", strerror(errno));
```

Comment:
do you need to keep the `goto err_close_shm_ep` in the error path?

Reply:
I think I understand now. does this look good?